fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/Loan.java

```diff
@@ -3510,6 +3510,39 @@ private void processIncomeAccrualTransactionOnLoanClosure() {
         updateLoanOutstandingBalances();
     }
 
+    public void applyIncomeAccrualTransaction(LocalDate closedDate) {
+        ExternalId externalId = ExternalId.empty();
+        boolean isExternalIdAutoGenerationEnabled = TemporaryConfigurationServiceContainer.isExternalIdAutoGenerationEnabled();
+        if (isPeriodicAccrualAccountingEnabledOnLoanProduct()) {
+            List<LoanTransaction> updatedAccrualTransactions = retrieveListOfAccrualTransactions();
+            LocalDate lastAccruedDate = this.getDisbursementDate();
+            if (!updatedAccrualTransactions.isEmpty()) {
+                lastAccruedDate = updatedAccrualTransactions.get(updatedAccrualTransactions.size() - 1).getTransactionDate();
+            }
+            HashMap<String, Object> feeDetails = new HashMap<>();
+            determineFeeDetails(lastAccruedDate, closedDate, feeDetails);
+            if (isExternalIdAutoGenerationEnabled) {
+                externalId = ExternalId.generate();
+            }
+            BigDecimal fee = (BigDecimal) feeDetails.get(FEE);
+            if (fee == null) {
+                fee = BigDecimal.ZERO;
+            }
+            BigDecimal penalty = (BigDecimal) feeDetails.get(PENALTY);
+            if (penalty == null) {
+                penalty = BigDecimal.ZERO;
+            }
+            BigDecimal total = fee.add(penalty);
+            // TODO: calculate interest?
```

Comment:
Please check whether we need the interest calculation as well, or do it later...

Reply:
We have comment this will be later